package/MDAnalysis/lib/_cutil.pyx

```diff
@@ -49,6 +49,70 @@ ctypedef cset[int] intset
 ctypedef cmap[int, intset] intmap
 
 
+@cython.boundscheck(False) # turn off bounds-checking for entire function 
+@cython.wraparound(False)  # turn off negative index wrapping for entire function 
+cdef np.intp_t[::1] _inverse_unique_unsorted_array(np.intp_t[::1] full_arr,
+                                          	   np.intp_t[::1] unique_arr,
+                                          	   np.intp_t[::1] mask):
+    """Find the inverse of a unique 1D array to map onto a non-unique 1D array.
+
+    Cython driver function. Loops though the unique array and finds the locations
+    in the full array that correspond to the values.
+    Requires contiguous arrays as an input.
+
+    Parameters
+    ----------
+    full_arr: numpy.ndarray
+        1D array of dtype ``numpy.int64`` in which to find the non-unique values.
+    unique_arr: numpy.ndarray
+        1D array of dtype ``numpy.int64``, the unique, unsorted values of full_arr.
+    mask: numpy.ndarray
+	1D array of dtype ``numpy.int64``, preallocated inverse array to be filled.
+
+    Returns
+    -------
+    numpy.ndarray
+        1D array of dtype ``numpy.int64``, the inverse that maps unique_arr onto full_arr.
+
+    """
+    
+    cdef Py_ssize_t i = 0 
```

Comment:
The use of `Py_ssize_t` here is correct and should really be done more in our cython layers. Just an observation.

Thanks for your contribution!